crates/holochain_websocket/src/test.rs

```diff
@@ -252,3 +252,73 @@ async fn ipv6_or_ipv4_connect() {
 
     l_task.await.unwrap();
 }
+
+#[tokio::test(flavor = "multi_thread")]
+#[ignore = "Requires a port to be free so should not run on CI"]
```

Comment:
We bind so many ports on CI and run parallel test runs for different PRs so even if I try to pick a free port there is no guarantee that this test will not be flaky... But it would be good to test that this works. Any ideas other than allowing it to be run locally?

Reply:
Has this test proven to be flaky on CI? If not, I prefer to keep it until it becomes flaky. Aren't there multiple tests with actual websocket connections that are not flaky on CI?

Reply:
The test on its own won't be flaky, it's because we use a shared builder. So as soon as two people run the test suite at the same time it will collide. We've run into this a few times with tests that bind specific ports :(

Reply:
test strategy to test binding to specific ports

if we restrict the test to linux, we can make use of network namespaces.

roughly it would work by:

1. [`fork`](https://docs.rs/fork/latest/fork/fn.fork.html)
2. in the child, [`unshare`](https://docs.rs/nix/latest/nix/sched/fn.unshare.html) with [`CLONE_NEWNET`](https://docs.rs/nix/latest/nix/sched/struct.CloneFlags.html#associatedconstant.CLONE_NEWNET) so it's in a separate network namespace.
3. in the child, run the test logic
4. in the parent, wait for the child to complete